internal/server/data/errors.go

```diff
@@ -0,0 +1,8 @@
+package data
+
+import "fmt"
+
+var (
+	ErrAccessKeyExpired          = fmt.Errorf("access key expired")
+	ErrAccessKeyDeadlineExceeded = fmt.Errorf("%w: extension deadline exceeded", ErrAccessKeyExpired)
```

Comment:
minor: I usually try to keep package-level variables like this in the same file where they are used. It makes it much easier to split up a package without a large diff, because files can be `git mv` instead of having to copy bits from multiple files.

I find grouping things by feature or domain instead of grouping by kind often works out well.